### PR TITLE
Adding check if dex-tls secrets is repeated many times

### PR DESCRIPTION
**Pull request by mmartin24**
PR for https://github.com/epinio/epinio/issues/2044

### Summary
I modified the place for the check. Originally it was thought in [acceptance-cluster-setup.sh](https://github.com/epinio/epinio/blob/f935d3cfdced05ba32b8e9535ab513257f52b9b2/scripts/acceptance-cluster-setup.sh) or  [acceptance-cluster-setup-kind.sh]( https://github.com/epinio/epinio/blob/f935d3cfdced05ba32b8e9535ab513257f52b9b2/scripts/acceptance-cluster-setup-kind.sh)  but I realized that the `cert-manager` pod is not set yet at this stage so I did it instead on the [prepare-environment-k3d.sh](https://github.com/epinio/epinio/blob/e7d671801630e0c383a45f0dcf9b72027bb48da3/scripts/prepare-environment-k3d.sh)

---
### Done
Added check on  cert-manager/cert-manager pod for log `"secret"="dex-tls" "message"="unexpected managed Secret Owner Reference field on Secret --enable-certificate-owner-ref=true"`

If successful the installation continues as normal
If found more than 10 times when installing epinio the test fails and displays:
- Reason of failure
- Counts of occurrences of that login

Error displayed:

```
-------------------------------------
Cleanup old settings
-------------------------------------
Trying to login. ✔️
-------------------------------------
Trying to getting info. ✔️
-------------------------------------

-------------------------------------
Warning: 'dex-tls' secrets may be be updated many times a second.
More than '3351' logs found in 'cert-manager/cert-manager' pod with entry = ' "secret"="dex-tls" "message"="unexpected managed Secret Owner Reference field on Secret --enable-certificate-owner-ref=true" '
Exiting installation
-------------------------------------
make: *** [Makefile:221: prepare_environment_k3d] Error 1
```
Note: 
To check it: deploy epinio with version `v1.6.1` . For instance add this after the `patch-epinio-deployment.sh` part of the original script: 

```
helm upgrade --install epinio -n epinio --version 1.6.1 --create-namespace epinio/epinio --set global.domain=172.19.0.3.omg.howdoi.website
```

### CODE REVIEW

**Review by andreas-kupries — DISMISSED** [review]

**Comment by mmartin24** [comment]
Just adapted test to check cert-manager targeting the label in case more pods are affected

**Review by andreas-kupries — DISMISSED** [review]

**Review by thehejik — APPROVED** [review]

**Comment by mmartin24** [comment]
Just detected an error that silently escaped to the tests here: https://github.com/epinio/epinio/actions/runs/4023449208/jobs/6914436461#step:10:327
```
./scripts/prepare-environment-k3d.sh: line 147: [: -gt: unary operator expected
```

In addition to this the check I did was using `zsh` and when using `bash ` it fails to read the line well:
```
mmartin@mmartin-thinkpad-p15-gen-2i:~/suse/epinio-repos/epinio$ message_dex_tls='"secret"="dex-tls" "message"="unexpected managed Secret Owner Reference field on Secret --enable-certificate-owner-ref=true"'
check_dex_log_count=$(kubectl logs  -n cert-manager -lapp=cert-manager --tail=-1 | grep ${message_dex_tls} | wc -l)
grep: unrecognized option '--enable-certificate-owner-ref=true"'
Usage: grep [OPTION]... PATTERNS [FILE]...
Try 'grep --help' for more information.
```

Fixing both

**Review by thehejik — APPROVED** [review]

File: scripts/prepare-environment-k3d.sh [hunk 1]
```diff
@@ -121,6 +121,7 @@ else
       { "backend": { "service": { "name": "epinio-server", "port": { "number": 80 } } }, "path": "/exit", "pathType": "ImplementationSpecific" } }]'
   fi
 fi
+helm upgrade --install epinio -n epinio --version 1.6.1 --create-namespace epinio/epinio --set global.domain=172.19.0.3.omg.howdoi.website
```
Comment (andreas-kupries) [inline scripts/prepare-environment-k3d.sh]: This looks like leftover debug code, with its fixed version and host.

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

File: scripts/prepare-environment-k3d.sh (hunk 1 above)
Comment (andreas-kupries) [reply scripts/prepare-environment-k3d.sh]: Never mind. Next commit removed it.

**Review by andreas-kupries — APPROVED** [review]

File: scripts/prepare-environment-k3d.sh (hunk 1 above)
Comment (mmartin24) [reply scripts/prepare-environment-k3d.sh]: yes. It was. 
thanks for checking

**Pull request merged**